lib/providers/profiles.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:flutter/widgets.dart';
+
+import '../models/profile.dart';
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:basic_authentication_flutter_challenge/models/http_exception.dart';
+import 'package:http/http.dart' as http;
+
+class Profiles with ChangeNotifier {
+  List<Profile> _profiles = [];
+  final String? authToken;
+  final int? userId;
+  Profile? mainProfile;
+
+  Profiles(this.authToken, this.userId, this._profiles, this.mainProfile);
+
+  List<Profile> get profiles {
+    return [..._profiles];
+  }
+
+  Profile get profile {
+    return mainProfile!;
+  }
+
+  Future<void> loggedinUserProfile() async {
+    final url = Uri.parse('http://161.35.99.225/api/v1/users/$userId');
+    try {
+      final response = await http.get(
```

Comment:
It is better to separate between business logic and fetching data. 
This `Profiles` class should care about business logic while fetching data form the backend should be the speciality of another class that is usually called `Repository` class. 
So one way to do this is to create a `ProfileRepository` class that deals with network requests and handle any network errors.

Reply:
created the separation between business and fetching logics